Fixed example Snippet and clarified it.

sas41 opened the pull request:
# Found a possible error in one of the snippets

## Summary

Fixes the 3rd snippet's return type and clarifies the local function's name to emphasize it's status as a local function

## Details

Changed return type of snippet 3 from bool to int, as returned variable was declared as int.

sas41 committed:
Fixed example Snippet and clarified it.

pkulikov commented on docs/csharp/local-functions-vs-lambdas.md:
````diff
@@ -65,20 +65,20 @@ Third, the compiler can perform static analysis that enables local functions to
 }
 ```
 
 The compiler can determine that `Local` definitely assigns `y` when called. Because `Local` is called before the `return` statement, `y` is definitiely
````
As `Local` is renamed to `LocalFunction` in the example, please update the paragraph after the example as well.

rpetrusha commented on docs/csharp/local-functions-vs-lambdas.md:
```diff
@@ -65,20 +65,20 @@ Third, the compiler can perform static analysis that enables local functions to
 assigned at the `return` statement.
 
-The analysis that enables that analysis enables the fourth difference.
+The analysis that enables the example analysis enables the fourth difference.
```
Even with your clarification of "that", the meaning of this sentence is unclear. I'd suggest removing it, and changing the next sentence to "Fourth, depending..."

rpetrusha reviewed (changes requested):
Thank you for contributing to the dotnet/docs repo, @sas41, and for clarifying both code and text. I've suggested a change to the paragraph that you modified. Also note the comment from @pkulikov about changing two occurrences of `Local` to `LocalFunction` in the sentence that follows the example.

sas41 committed:
Renamed "Local" to "LocalFunction"

From my previous commit, pointed out by @pkulikov, I forgot to rename the later mentions of the "Local" function in the paragraph after the code snippet.

rpetrusha commented:
Thank you so much, @sas41, for making the additional change. We'll merge your PR now. Your changes should appear live on docs.microsoft.com in the next day or two.
